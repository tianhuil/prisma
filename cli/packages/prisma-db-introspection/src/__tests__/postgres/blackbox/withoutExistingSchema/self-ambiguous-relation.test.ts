import testSchema from '../common'

describe('Introspector', () => {
  // Schema from https://github.com/prismagraphql/prisma/issues/2504
  test('test schema - 2504', async () => {
    await testSchema(sql)
  })
})

const sql = `
/*******************************************************************************
 Create Tables
********************************************************************************/

CREATE TABLE "Employee"
(
  "EmployeeId" INT NOT NULL,
  "LastName" VARCHAR(20) NOT NULL,
  "FirstName" VARCHAR(20) NOT NULL,
  "Title" VARCHAR(30),
  "ReportsTo" INT,
  "BirthDate" TIMESTAMP,
  "HireDate" TIMESTAMP,
  "Address" VARCHAR(70),
  "City" VARCHAR(40),
  "State" VARCHAR(40),
  "Country" VARCHAR(40),
  "PostalCode" VARCHAR(10),
  "Phone" VARCHAR(24),
  "Fax" VARCHAR(24),
  "Email" VARCHAR(60),
  CONSTRAINT "PK_Employee" PRIMARY KEY  ("EmployeeId")
);

/*******************************************************************************
 Create Primary Key Unique Indexes
********************************************************************************/

/*******************************************************************************
 Create Foreign Keys
********************************************************************************/

ALTER TABLE "Employee" ADD CONSTRAINT "FK_EmployeeReportsTo"
  FOREIGN KEY ("ReportsTo") REFERENCES "Employee" ("EmployeeId") ON DELETE NO ACTION ON UPDATE NO ACTION;

CREATE INDEX "IFK_EmployeeReportsTo" ON "Employee" ("ReportsTo");
`
