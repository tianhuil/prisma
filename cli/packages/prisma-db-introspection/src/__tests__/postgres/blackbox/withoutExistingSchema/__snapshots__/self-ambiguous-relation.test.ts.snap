// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Introspector test schema - 2504 1`] = `
"type Employee {
  id: Int! @pgColumn(name: \\"EmployeeId\\") @unique
  address: String @pgColumn(name: \\"Address\\")
  birthDate: DateTime @pgColumn(name: \\"BirthDate\\")
  city: String @pgColumn(name: \\"City\\")
  country: String @pgColumn(name: \\"Country\\")
  email: String @pgColumn(name: \\"Email\\")
  employee: [Employee]
  fax: String @pgColumn(name: \\"Fax\\")
  firstName: String! @pgColumn(name: \\"FirstName\\")
  hireDate: DateTime @pgColumn(name: \\"HireDate\\")
  lastName: String! @pgColumn(name: \\"LastName\\")
  phone: String @pgColumn(name: \\"Phone\\")
  postalCode: String @pgColumn(name: \\"PostalCode\\")
  reportsTo: Employee @pgColumn(name: \\"ReportsTo\\")
  state: String @pgColumn(name: \\"State\\")
  title: String @pgColumn(name: \\"Title\\")
}"
`;

exports[`Introspector test schema - 2504 2`] = `
"type Employee {
  EmployeeId: Int! @id
  address: String @db(name: \\"Address\\")
  birthDate: DateTime @db(name: \\"BirthDate\\")
  city: String @db(name: \\"City\\")
  country: String @db(name: \\"Country\\")
  email: String @db(name: \\"Email\\")
  employee: [Employee]
  fax: String @db(name: \\"Fax\\")
  firstName: String! @db(name: \\"FirstName\\")
  hireDate: DateTime @db(name: \\"HireDate\\")
  lastName: String! @db(name: \\"LastName\\")
  phone: String @db(name: \\"Phone\\")
  postalCode: String @db(name: \\"PostalCode\\")
  reportsTo: Employee @db(name: \\"ReportsTo\\")
  state: String @db(name: \\"State\\")
  title: String @db(name: \\"Title\\")
}"
`;
